src/turbopuffer/_utils/_transform.py

```diff
@@ -387,37 +401,21 @@ async def _async_transform_recursive(
     """
     from .._compat import model_dump
 
-    # Fast path for primitive types that never need transformation.
-    # This avoids expensive type introspection for the most common cases.
-    if data is None or isinstance(data, (int, float, bool, str)):
-        return data
+    if inner_type is None:
+        inner_type = annotation
 
-    # Fast path for vector encoding - must come BEFORE list-of-primitives check
-    # because vectors are lists of floats that need base64 encoding.
+    # Fast path for vector encoding.
     if annotation in VectorAnnotations:
         return _encode_vector(data)
```

Comment:
The intent is just to put this function back to the way it was, right? If so can we remove this fast path for vector encoding, so that it exactly matches the default Stainless codegen?